rename deprecated parametrs

b4ldr opened the pull request:
This is to address #24

xaque208 commented on manifests/init.pp:
```diff
@@ -73,7 +73,7 @@
   # portstree management has moved to poudriere::portstree
   if $cron_enable == true {
-    notice('cron_enable, cron_interval and port_fetch_method on class poudriere is deprecated, define seperately poudriere::portstree')
+    notice('cron_enable, cron_interval and port_fetch_method on class poudriere is deprecated, define seperately poudriere::portstree.  for the default portdstreee us default_${parameter} e.g. default_cron_enable')
```
A couple typos in this line.

xaque208 commented:
The term default here to me implies that if I don't set those parameters on the `poudriere::portstree` class, that those defaults will be used instead.  That's not the case here though, and the 'defaults' are being used to build out the ports tree named 'default' with those parameters.  I'm not yet convinced this is the right course of action, though I could be convinced.

Would it perhaps make more sense to remove the 'default' ports tree from the `poudriere` class entirely, update the readme, and force the users to do as the deprecation warning has said to do?  That has the potential to confused users who have missed the deprecation warning now that they would have a default ports tree on their systems, but it would no longer be managed by puppet.

b4ldr commented:
Personally i like the idea to force the user to create the portstree them self.  When i first started using the module that is how i expected it to work and found it somewhat confusing enabling the cron job on the default portstree.  I am also not a fan of the default_* parameters and changing them to default_portstree_* looks horrible.  

Changing now would cause confusion to users that upgrade perhaps you could add a warning if the default portstree is not defined like

```Puppet
if $warn_missing_portstree
  if ! defined Poudriere::Portstree['default'] {
    notice('the "default" portstree is no longer defined by this module please define it your self.  if you dont need the default portstree please set warn_missing_portstree to false')
  }
}
```

xaque208 commented:
I'd agree with that, though I'd like to see something in the readme about updating manifests to the new version with an updated example.  I'd be fine with removing the default ports tree management entirely as well with your warning that directs the user to the readme where their questions will be answered.

b4ldr commented:
Ok i have removed the default_* params and placed a check in the env  class to warn if the portstree passed to poudriere::env is not defined.  It seems to make more sense here as it also captures passing a custom portstree that does not exist.   The warning is slightly different if the portstree passed is named default to try and reflect the change in the behaviour.   i have also updated the readme.  

I toyed with the idea of making this check a fail or a warning and decided to go for warning as it is possible that people may manage the portstree outside of puppet.  let me know what you think

xaque208 commented:
This looks good, thank you.
